wallet/src/main/scala/org/bitcoins/wallet/models/IncomingTransactionTable.scala

```diff
@@ -0,0 +1,134 @@
+package org.bitcoins.wallet.models
+
+import org.bitcoins.core.crypto._
+import org.bitcoins.core.number.UInt32
+import org.bitcoins.core.protocol.script.EmptyScriptSignature
+import org.bitcoins.core.protocol.transaction.{
+  BaseTransaction,
+  EmptyWitness,
+  Transaction,
+  TransactionInput,
+  WitnessTransaction
+}
+import slick.jdbc.SQLiteProfile.api._
+import slick.lifted.{PrimaryKey, ProvenShape}
+
+/** Represents a relevant transaction for the wallet that we should be keeping track of */
+case class IncomingTransactionDb(
+    txIdBE: DoubleSha256DigestBE,
+    transaction: Transaction,
+    unsignedTxIdBE: DoubleSha256DigestBE,
+    unsignedTx: Transaction,
+    numInputs: Int,
+    numOutputs: Int,
+    lockTime: UInt32) {
```

Comment:
I think we should have `incomingAmount` or something like that as well. Also thoughts on adding `wtxid: Option[DoubleSha256DigestBE]` and and `tx_type` that can be `Base`, `SegwitV0` etc? Also do we want an `outputIndex`, so we know which output is ours at the database level? 

Finally, do we want to try and support multiple outputs paying to us in the same tx (batching)? I understand if we don't want to support this for now, but please file an issue if not.

Reply:
Also, with `IncomingTransactions` we don't really necessarily have access to the `unsignedTxId` do we? That is something that is only applicable for outgoing txs no?

EDIT: Actually thinking about this more, maybe this can be a malleabillity prevention mechanism for `BaseTransaction`? I don't know, we should discuss more. We need to be _REALLY_ careful about handling malleability with `BaseTransaction`.